### PR TITLE
docs: Update docs for DELETE support

### PR DESCRIPTION
**Pull request by akshatnair**
DELETE command has some restrictions that needed documenting.

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/delete.rst [hunk 1]
```diff
@@ -30,3 +30,8 @@ Delete all line items for low priority orders::
 Delete all orders::
 
     DELETE FROM orders;
+
+Limitations
+-------------
+Only :doc:`../connector/hive` currently supports DELETE and it works for the queries where one or more partitions are
```
Comment (martint) [inline presto-docs/src/main/sphinx/sql/delete.rst]: This is not accurate. The raptor connector also supports DELETE.

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -151,3 +151,8 @@ Hive, this table can be described in Presto::
 This table can then be queried in Presto::
 
     SELECT * FROM hive.web.page_view;
+
+Hive Connector Limitations
+---------------------------
+
+* :doc:`/sql/delete` : Only supports ``DELETE`` where one or more partitions are deleted entirely using condition on partitioning columns.
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/hive.rst]: I'd rephrase as "Only supports DELETE where the WHERE clause matches entire partitions"

File: presto-docs/src/main/sphinx/sql/delete.rst (hunk 1 above)
Comment (electrum) [inline presto-docs/src/main/sphinx/sql/delete.rst]: We should remove this entirely as this is the SQL documentation, not the Hive connector documentation.

Perhaps a more generic note like "some connectors may not support this operation or have restricted support, see the connector documentation for details".

Or should we add connector specific docs to all the various DDL and DML statements? Not sure how I feel about that.

File: presto-docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (akshatnair) [inline presto-docs/src/main/sphinx/connector/hive.rst]: The reason I had to phrase it like this is due to the limitations that if the where clause is based on a function of partitioning key then it does not delete even if it matches the entire partitions. Here is an example,
``
presto:default> show partitions from t1_p;
     c2     
------------
 1992-01-01 
 2013-02-01 
 2020-02-03 
(3 rows)

presto:default> select month(cast(c2 as date)) from t1_p;
 _col0 
-------
     2 
     1 
     2 
presto:default> delete from t1_p where month(cast(c2 as date))= 1;
Query 20160403_224232_00011_ppfuv failed: This connector only supports delete where one or more partitions are deleted entirely
``

File: presto-docs/src/main/sphinx/sql/delete.rst (hunk 1 above)
Comment (akshatnair) [inline presto-docs/src/main/sphinx/sql/delete.rst]: I will rephrase this and we can add connector specific limitations on the connector page

File: presto-docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (martint) [inline presto-docs/src/main/sphinx/connector/hive.rst]: Well, that's a bug. That should work, since the filter can be evaluated entirely based on the partition column.

Comment (akshatnair) [inline presto-docs/src/main/sphinx/connector/hive.rst]: Okay I will file an issue for this

**Comment by akshatnair** [comment]
updated

File: presto-docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (martint) [inline presto-docs/src/main/sphinx/connector/hive.rst]: https://github.com/prestodb/presto/issues/4932

File: presto-docs/src/main/sphinx/sql/delete.rst [hunk 3]
```diff
@@ -30,3 +30,7 @@ Delete all line items for low priority orders::
 Delete all orders::
 
     DELETE FROM orders;
+
+Limitations
+-------------
+Some connectors have limited or no support for DELETE, see the connector documentation for more details.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/sql/delete.rst]: Add newline after header

Comment (electrum) [inline presto-docs/src/main/sphinx/sql/delete.rst]: This should either be separated with a semicolon or be two separate sentences.

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 4]
```diff
@@ -153,3 +153,8 @@ Hive, this table can be described in Presto::
 This table can then be queried in Presto::
 
     SELECT * FROM hive.web.page_view;
+
+Hive Connector Limitations
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive.rst]: This section can just be "Limitations" since this is already the Hive connector docs

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 5]
```diff
@@ -153,3 +153,8 @@ Hive, this table can be described in Presto::
...
+---------------------------
+
+* :doc:`/sql/delete` : Only supports ``DELETE`` if the WHERE clause matches entire partitions.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive.rst]: This formatting is a bit weird. Since we only have one item now, it should just be a normal sentence. We can figure out a different way to structure it later when there are multiple items.

**Comment by electrum** [comment]
Commit message: we don't include things like "docs" at the beginning

**Comment by electrum** [comment]
We can simplify the commit message to "Document restrictions for DELETE"

**Comment by electrum** [comment]
Merged, thanks!
